projects/solar/test/test_fault_monitor.c

```diff
@@ -3,17 +3,41 @@
 #include <stdbool.h>
 
 #include "data_store.h"
-#include "event_queue.h"
+#include "exported_enums.h"
 #include "log.h"
 #include "ms_test_helpers.h"
 #include "solar_events.h"
 #include "test_helpers.h"
 #include "unity.h"
 
+#define MAX_FAULTS 32
+
 #define TEST_OUTPUT_OVERCURRENT_THRESHOLD 1000000000L
 #define TEST_OUTPUT_OVERVOLTAGE_THRESHOLD 1000000000uL
 #define TEST_TEMPERATURE_THRESHOLD 1000000000uL
 
+#define TEST_ASSERT_NO_FAULT() TEST_ASSERT_EQUAL(0, s_num_faults_raised)
+
+#define TEST_ASSERT_SINGLE_FAULT(fault, data)       \
+  ({                                                \
+    TEST_ASSERT_EQUAL(1, s_num_faults_raised);      \
+    TEST_ASSERT_EQUAL((fault), s_faults_raised[0]); \
+    TEST_ASSERT_EQUAL((data), s_fault_data[0]);     \
+    s_num_faults_raised = 0;                        \
+  })
```

Comment:
This pattern is used in many other places (e.g. `ms_test_helpers.h`, `can_transmit.h`, `misc.h`) and ~8 times more frequently than `do {} while (0)`.

Reply:
Did some research and apparently it's not listed on the style guide. do/while is nice because this pattern is a gcc-only thing but we should probz stick with what's in the can_transmit autogen code lol. 

@g-jessmuir we might want to cleanup the do/while macros if any and add to the style guide I guess.